app/models/patient.rb

```diff
@@ -32,10 +32,16 @@ class Patient
   mongoid_userstamp user_model: 'User'
 
   # Methods
+
+
   def self.search(name_or_phone_str) # TODO: optimize
-    name_matches = Patient.where name: name_or_phone_str
-    primary_matches = Patient.where primary_phone: name_or_phone_str
-    secondary_matches = Patient.where secondary_phone: name_or_phone_str
-    (name_matches | primary_matches | secondary_matches)
+    clean_phone = name_or_phone_str.gsub(/\D/, '')
+    phone = clean_phone[0..2] + '-' + clean_phone[3..5] + '-' + clean_phone[6..10]
+    begin
+      name_matches = Patient.where ({name: /^#{Regexp.escape(name_or_phone_str)}$/i})
```

Comment:
@colinxfleming Happy to make these changes! I also found once I upgraded rails that search was throwing an error unless I converted these parts (ex: clean_phone[0..2]) to a string with the to_s method. Additionally I have a solution for the chronological order of the change log for issue #403 - should I submit that along with this item or would you prefer that in its own PR?